File: src/sims/registry.ts
import type { SimDef } from './types';
import { notesSim } from './notes';
import { imessageSim } from './imessage';
import { emailSim } from './email';

/* Roadmap placeholders for the sims not yet rebuilt — they show in the launcher
   as "soon" so the full roster is visible. Each becomes a real entry (ready:true
   with a Component) as it's ported from /legacy, one at a time. */
function soon(id: string, label: string, glyph: string, accent: string, frame: SimDef['frame']): SimDef {
  return { id, label, glyph, accent, frame, logical: { w: 1080, h: 1920 }, ready: false, defaultScript: '' };
}

export const SIMS: SimDef[] = [
  notesSim,
  imessageSim,
  emailSim,
  soon('lists', 'Lists', '✅', '#5b8def', 'phone'),
  soon('corporate', 'Corporate', '🏢', '#8a94a6', 'desktop'),
  soon('typer', 'Typer', '⌨️', '#a78bfa', 'free'),
  soon('tiktok', 'TikTok', '🎵', '#ff2d55', 'phone'),
  soon('claude', 'Claude', '✳️', '#d97757', 'free'),
  soon('journal', 'Journal', '📔', '#c0875b', 'phone'),
];

export function getSim(id: string): SimDef | undefined {
  return SIMS.find((s) => s.id === id);
}

export function defaultSimId(): string {
  const stored = (() => { try { return localStorage.getItem('hub:current'); } catch { return null; } })();
  if (stored && getSim(stored)?.ready) return stored;
  return SIMS.find((s) => s.ready)?.id ?? SIMS[0].id;
}

File: app/src/main/java/com/example/recorder/sims/Registry.kt
package com.example.recorder.sims

import com.example.recorder.sims.claude.ClaudeSim
import com.example.recorder.sims.email.EmailSim
import com.example.recorder.sims.imessage.IMessageSim
import com.example.recorder.sims.lists.ListsSim
import com.example.recorder.sims.notes.NotesSim
import com.example.recorder.sims.typer.TyperSim
import com.example.recorder.sims.typewriter.TypewriterSim
import com.example.recorder.sims.whatsapp.WhatsAppSim

/**
 * The sim registry. Port of src/sims/registry.ts. (Corporate folded into Notes
 * as a Document type; TikTok skipped. Remaining: journal.)
 */
val SIMS: List<SimDef> = listOf(
    NotesSim,
    IMessageSim,
    WhatsAppSim,
    EmailSim,
    ListsSim,
    TyperSim,
    TypewriterSim,
    ClaudeSim,
)

fun getSim(id: String?): SimDef? = SIMS.firstOrNull { it.id == id }

fun defaultSim(): SimDef = SIMS.firstOrNull { it.ready } ?: SIMS.first()
